src/test/java/teammates/test/cases/webapi/DeleteFeedbackResponseActionTest.java

```diff
@@ -20,16 +41,232 @@ protected String getRequestMethod() {
+    }
+
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+        verifyHttpParameterFailure(Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString());
+
+        ______TS("Typical success case, student");
+
+        String[] params = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        DeleteFeedbackResponseAction a = getAction(params);
+        JsonResult r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        // response is deleted
+        assertNull(logic.getFeedbackResponse(typicalResponse.getId()));
+
+        ______TS("Typical success case, instructor");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        params = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse3.getId(),
+        };
+
+        a = getAction(params);
+        r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        // response is deleted
+        assertNull(logic.getFeedbackResponse(typicalResponse3.getId()));
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
-        // TODO
+
+        ______TS("wrong giver type");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+        String[] wrongGiverTypeParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        verifyCannotAccess(wrongGiverTypeParams);
+
+        ______TS("preview mode, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] previewParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+                Const.ParamsNames.PREVIEWAS, instructor1OfCourse1.email,
+        };
+
+        verifyCannotAccess(previewParams);
+
+        ______TS("response in session not open, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] sessionNotOpenParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, responseInClosedSession.getId(),
+        };
+
+        verifyCannotAccess(sessionNotOpenParams);
+
+        ______TS("Response contains question not intended shown to instructor, "
```

Comment:
Verify this? `Response contains question not intended shown to instructo, moderated instructor should not be accessible`